app/util/transactions/index.js

```diff
@@ -127,7 +127,7 @@ const actionKeys = {
  * @param {Object} opts - Optional asset parameters
  * @returns {String} - String containing the generated transfer data
  */
-export function generateTransferData(type, opts) {
+export function generateTransferData(type = undefined, opts = {}) {
```

Comment:
was going to suggest renaming to `.ts` (like the test file), then

```suggestion
export function generateTransferData(
  type = undefined,
  opts = {},
): asserts type {
```

but the rename exposes _a lot_ more type errors 😅

so noted it here as a worthy long-term follow-up
- https://github.com/MetaMask/metamask-mobile/issues/5993